src/k8s/pkg/k8sd/features/interface.go

```diff
@@ -0,0 +1,65 @@
+package features
+
+import (
+	"context"
+
+	"github.com/canonical/k8s/pkg/k8sd/types"
+	"github.com/canonical/k8s/pkg/snap"
+)
+
+// Interface abstracts the management of built-in Canonical Kubernetes features.
+type Interface interface {
```

Comment:
Go best-practice would probably be to call this `Featurer` but I agree that this is weird. `feature.Interface` is fine.